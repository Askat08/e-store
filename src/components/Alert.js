import React from "react";

const style = {
  msg: {
    position: "fixed",
    top: "10px",
    right: "",
    fontSize: "2rem",
    background: "lightgreen",
    width: "100%",
    zIndex: "100",
    textAlign: "center",
    opacity: "0.9",
    transition: "opacity 0.1s ease-out",
  },
};

export default function Alert() {
  return (
    <div style={style.msg}>
      <h3 className="lead">Hooray! Comic book in your cart.</h3>
    </div>
  );
}
