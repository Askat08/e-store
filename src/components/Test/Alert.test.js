import React from "react";
import { shallow } from "enzyme";
import Alert from "../Alert";

describe("Alert component test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Alert />);
  });
  it("should render Alert component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have same style for div", () => {
    const style = wrapper.find("div").get(0).props.style;
    expect(style).toHaveProperty("position", "fixed");
    expect(style).toHaveProperty("top", "10px");
    expect(style).toHaveProperty("right", "");
    expect(style).toHaveProperty("fontSize", "2rem");
    expect(style).toHaveProperty("background", "lightgreen");
    expect(style).toHaveProperty("width", "100%");
    expect(style).toHaveProperty("zIndex", "100");
    expect(style).toHaveProperty("textAlign", "center");
    expect(style).toHaveProperty("opacity", "0.9");
    expect(style).toHaveProperty("transition", "opacity 0.1s ease-out");
  });

  it("should test h3 text", () => {
    expect(wrapper.find("h3").text()).toEqual(
      "Hooray! Comic book in your cart."
    );
  });
});
