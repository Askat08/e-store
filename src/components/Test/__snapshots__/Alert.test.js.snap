// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component test should render Alert component 1`] = `
<div
  style={
    Object {
      "background": "lightgreen",
      "fontSize": "2rem",
      "opacity": "0.9",
      "position": "fixed",
      "right": "",
      "textAlign": "center",
      "top": "10px",
      "transition": "opacity 0.1s ease-out",
      "width": "100%",
      "zIndex": "100",
    }
  }
>
  <h3
    className="lead"
  >
    Hooray! Comic book in your cart.
  </h3>
</div>
`;
